robosherlock/src/flowcontrol/src/run.cpp

```diff
@@ -148,10 +151,14 @@ int main(int argc, char *argv[])
 
   try
   {
-    RSProcessManager manager(useVisualizer, waitForServiceCall, keType, save_path);
-    manager.setUseIdentityResolution(useObjIDRes);
-    manager.init(analysis_engine_file, pervasive, parallel);
-    manager.run();
+    //create the perception system
+    RSProcessManager* manager=new RSProcessManager(useVisualizer, waitForServiceCall, keType, save_path);
+    manager->setUseIdentityResolution(useObjIDRes);
+    manager->init(analysis_engine_file, pervasive, parallel);
+    //create,start the query action server and pass the rosnode and perception system to it
+    RSSimpleQueryActionServer actionServer(nh,manager);
+    //start the perception system
+    manager->run();

```

Comment:
We should not have to change the initialization of RSProcessManager. RSSimpleQueryActionServer shoudl be a member of the RSProcessManager class, and be initialized in the constructor. 

Another note here: you made a pointer out of ``manager``, allocated memory for it, but never delete it.

Reply:
done